handler/areas.go

```diff
@@ -0,0 +1,69 @@
+package handler
+
+import (
+	"net/http"
+
+	"github.com/ONSdigital/dp-api-clients-go/v2/cantabular"
+	"github.com/ONSdigital/dp-cantabular-dimension-api/contract"
+	"github.com/ONSdigital/dp-cantabular-dimension-api/model"
+	dperrors "github.com/ONSdigital/dp-net/v2/errors"
+	"github.com/gorilla/schema"
+
+	"github.com/pkg/errors"
+)
+
+// Areas handles requests to /area-types
+type Areas struct {
+	respond responder
+	ctblr   cantabularClient
+}
+
+// NewAreas returns a new Areas handler
+func NewAreas(r responder, c cantabularClient) *Areas {
+	return &Areas{
+		respond: r,
+		ctblr:   c,
+	}
+}
```

Comment:
I see this quite often, so I am not sure if this is idiomatic Go at ONS.
But what is the advantage of a pointer over a copy of Areas?

In this case I believe the performance, although negligibly, is impacted, because now the GC needs to keep a track of reference in the `heap`, instead of just copying an instance and putting it on the `stack` of the next function.

Looking at where `NewAreas(...)` is being called I cannot see us trying to evaluate the zero-value of it, which would have been `nil`, one of the other reasons why we may have wanted to return a pointer.

To my mind, I would suggest: 

```suggestion
func NewAreas(r responder, c cantabularClient) Areas {
	return Areas{
		respond: r,
		ctblr:   c,
	}
}
```

As I pointed out, this is negligible point as I see a lot of code like this, but I am more curious on the _why_.
I would also only encourage the use of pointers if needed, as observed by Jon Bodner's book [Learning Go](https://www.amazon.co.uk/Learning-Go-Idiomatic-Real-World-Programming/dp/1492077216).

Anyway... Friday afternoon and I have nothing better to do... :roll_eyes: :smile: